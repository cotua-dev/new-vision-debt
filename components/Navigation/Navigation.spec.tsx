import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react';
import { Navigation } from './index';

describe('Navigation Component', (): void => {
    let renderResult: RenderResult;

    beforeEach((): RenderResult => renderResult = render(<Navigation/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    test('clicking hamburger activates mobile navigation', (): void => {
        const { queryByTestId } = renderResult;

        // Set the innerWidth to mobile
        global.innerWidth = 1000;

        // Click on the mobile menu button
        const mobileMenuButton = queryByTestId('open-mobile-navigation-button');
        fireEvent.click(mobileMenuButton);

        // Check to see if the mobile menu is truthy
        const mobileMenu = queryByTestId('mobile-navigation');
        expect(mobileMenu).toBeTruthy();

        // Click on the close mobile menu button
        const closeMobileMenuButton = queryByTestId('close-mobile-navigation-button');
        fireEvent.click(closeMobileMenuButton);

        // Reset innerWidth to desktop
        global.innerWidth = 1080;
    });

    afterEach(cleanup);
});
