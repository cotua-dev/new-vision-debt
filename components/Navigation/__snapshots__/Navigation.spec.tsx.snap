// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders 1`] = `
<nav
  class="debt-advisors-navigation"
>
  <a
    aria-label="Home Page Link"
    data-testid="nav-logo-link"
    href="/"
  >
    <picture>
      <source />
      <img
        alt="Debt Advisors Of America Logo"
        class="logo"
        data-testid="regular-logo"
        loading="lazy"
      />
    </picture>
  </a>
</nav>
`;
